### PR TITLE
Fix 1353

### LINKED ISSUE #1353: Feature: Create `rpt_site_details` using `Make_Snapshot()`
## Feature Details

For development, just add this to Make_Snapshot() output without modifying other outputs. We can remove unnecessary tables as we continue development.

E.g., The output of `Make_Snapshot()` should remain as is + adding `rpt_site_details`.

This should be its own function that is used within `Make_Snapshot()`.

### Inputs:
- `status_site`
- `results_summary`

### Output:

| Columns               | Data Type              | Source/Transformation                                                                             |
| --------------------- | ---------------------- | ------------------------------------------------------------------------------------------------- |
| study_id              | text                   |                                                                                                   |
| snapshot_date         | date                   |                                                                                                   |
| site_id               | text                   |                                                                                                   |
| site_nm               | text                   |                                                                                                   |
| site_status           | character varying(200) |                                                                                                   |
| investigator_nm       | text                   |                                                                                                   |
| site_country          | character varying(100) |                                                                                                   |
| site_state            | character varying(200) |                                                                                                   |
| site_city             | text                   |                                                                                                   |
| region                | character varying(100) | "Other"                                                                                           |
| enrolled_participants | integer                |                                                                                                   |
| planned_participants  | integer                | NULL                                                                                              |
| num_of_at_risk_kris   | integer (NOT NULL)     | Total number of distinct groups from results_summary where the flag is 1 0r -1 for a KRI workflow |
| num_of_flagged_kris   | integer (NOT NULL)     | Total number of distinct groups from results_summary where the flag is 2 0r -2 for a KRI workflow |
| pt_cycle_id           | character varying(200) |                                                                                                   |
| pt_data_dt            | character varying(200) |                                                                                                   |

### PR DESCRIPTION
**Pull request by andersonjake1988**
## Overview
fix #1353

## Test Notes/Sample Code

Notes:

### CODE REVIEW

**Comment by andersonjake1988** [comment]
[rpt_site_details.csv](https://github.com/Gilead-BioStats/gsm/files/13181543/rpt_site_details.csv)

File: R/util-MakeSnapshotHelpers.R [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    {if(!"amber" %in% names(.)) tibble::add_column(., "amber" = NA) else .} %>%
+    {if(!"red" %in% names(.)) tibble::add_column(., "red" = NA) else .} %>%
+    {if(group == "site") {select(.,"siteid" = "GroupID",
+                                 "num_of_at_risk_kris" = "amber",
+                                 "num_of_flagged_kris" = "red")}
+      else .} %>%
+    {if(group == "kri") {select(., "kri_id" = "kri",
+                                "num_of_sites_at_risk" = "amber",
+                                "num_of_sites_flagged" = "red")}
+      else .}
```
Comment (mattroumaya) [inline R/util-MakeSnapshotHelpers.R:65]: I reviewed your PRs out of order, but think that we should refactor this part slightly to make it more concise/easily read.

**Review by mattroumaya — APPROVED** [review]
Looks good! 

Will add a comparison for @nkalkur-gilead's review

**Comment by mattroumaya** [comment]
|     | imported_column_name  | imported_schema                               | gsm_name              | gsm_schema |
| --- | --------------------- | --------------------------------------------- | --------------------- | ---------- |
| 0   | study_id              | pyarrow.Field<study_id: string>               | study_id              | character  |
| 1   | snapshot_date         | pyarrow.Field<snapshot_date: date32[day]>     | snapshot_date         | Date       |
| 2   | site_id               | pyarrow.Field<site_id: string>                | site_id               | character  |
| 3   | site_nm               | pyarrow.Field<site_nm: string>                | site_nm               | character  |
| 4   | site_status           | pyarrow.Field<site_status: string>            | site_status           | character  |
| 5   | investigator_nm       | pyarrow.Field<investigator_nm: string>        | investigator_nm       | character  |
| 6   | site_country          | pyarrow.Field<site_country: string>           | site_country          | character  |
| 7   | site_state            | pyarrow.Field<site_state: string>             | site_state            | character  |
| 8   | site_city             | pyarrow.Field<site_city: string>              | site_city             | character  |
| 9   | region                | pyarrow.Field<region: string>                 | region                | character  |
| 10  | enrolled_participants | pyarrow.Field<enrolled_participants: int32>   | enrolled_participants | integer    |
| 11  | planned_participants  | pyarrow.Field<planned_participants: double>   | planned_participants  | numeric    |
| 12  | num_of_at_risk_kris   | pyarrow.Field<num_of_at_risk_kris: int32>     | num_of_at_risk_kris   | integer    |
| 13  | num_of_flagged_kris   | pyarrow.Field<num_of_flagged_kris: int32>     | num_of_flagged_kris   | integer    |
| 14  | pt_cycle_id           | pyarrow.Field<pt_cycle_id: string>            | pt_cycle_id           | character  |
| 15  | pt_data_dt            | pyarrow.Field<pt_data_dt: string>             | pt_data_dt            | character  |
| 16  | gsm_analysis_date     | pyarrow.Field<gsm_analysis_date: date32[day]> | gsm_analysis_date     | Date       |
|     |                       |                                               |                       |            |

**Pull request merged**